src/cocotb_bus/bus.py

```diff
@@ -47,19 +47,31 @@ def __init__(self, entity, name, signals, optional_signals=[], bus_separator="_"
                 See the *signals* argument above for details.
             bus_separator (str, optional): Character(s) to use as separator between bus
                 name and signal name. Defaults to '_'.
+            case_insensitive (bool, optional): Perform case-insensitive match on signal names.
+                Defaults to True.
             array_idx (int or None, optional): Optional index when signal is an array.
         """
         self._entity = entity
         self._name = name
         self._signals = {}
 
+        if case_insensitive:
+            signal_names = dir(self._entity)
+            signal_name_mapping = dict([(n.casefold(), n) for n in signal_names])
```

Comment:
No need for a temporary.
```suggestion
            signal_name_mapping = dict((n.casefold(), n) for n in signal_names)
```

Reply:
Sorry I'm no GitHub expert, but I get the feeling I should be able to click something here and accept changes. There is only a resolve conversation button though. This may be because I don't have write permissions to this repository?

Reply:
You would be making changes in your own branch. If you go into the "Files changed" tab, you should be able to see each comment and there should be a "commit suggestion" button.

Reply:
It looks like I managed to fork the repo with my personal account, and make the PR with my company account. No sure how that was possible, but I'll try and untangle it.